### PR TITLE
megadriv.xml: Added and corrected info on the Sega Game Toshokan games

### PR DESCRIPTION
**Pull request by ICEknigh7**
- Added list of known missing Sega Game Toshokan games.
- Added the specific titles of the collections where some of these dumps come from.
- Corrected Kandume -> Kanzume.
- Added proper years for all of their entries according to Sega's website.
- Added full "serialized" title to existing Riddle Wired entry.

### CODE REVIEW

**Comment by andlabs** [comment]
In addition:

Riddle Wired was a serialized release with new sets of questions every iteration. [This video](https://www.youtube.com/watch?v=g0DHI_m9xIg) shows a version from June; the one we have is for August. Both say "vol.1" but it seems like volumes were released weekly? (Sega Retro still needs to be updated.)

I can't tell if [Sega Net News](https://www.youtube.com/watch?v=YQuqYwZOnEY) was part of the Game Toshokan ROM or not.

Sega Ages 2500 Phantasy Star Complete Collection for PS2 comes with an alternate ROM for Nei's Adventure for some reason.

**Comment by ICEknigh7** [comment]
> Riddle Wired was a serialized release with new sets of questions every iteration. [This video](https://www.youtube.com/watch?v=g0DHI_m9xIg) shows a version from June; the one we have is for August. Both say "vol.1" but it seems like volumes were released weekly? (Sega Retro still needs to be updated.)

Very interesting, I'll add those too. I've just found a recent post with what seems to be a chronological list of all Game Toshokan releases (including something called "Formula Club" which will have to be investigated some day): https://mevius.5ch.net/test/read.cgi/gsaloon/1613901228/102-n/?v=pc

**Comment by ICEknigh7** [comment]
> Sega Ages 2500 Phantasy Star Complete Collection for PS2 comes with an alternate ROM for Nei's Adventure for some reason.

Ah, I see they even mentioned some changes being made in their official website: https://ages.sega.jp/vol32/ps2text.html

> (Since this work is a reproduction of the original version, some specifications are different from the ported version included in "Game no Kanzume" Vol.1 and Vol.2 for Mega CD)

I'll include it in the list of missing dumps for now.

File: hash/megadriv.xml [hunk 1]
```diff
@@ -60,6 +60,51 @@ Known undumped pirates
   * Shizen Mahjong / Competition Mahjong (by Sachen, apparently dumped but hoarded)
   * Magic 7 Block - 百変七巧板
 
+Sega Game Toshokan games could be downloaded via the Mega Modem while a subscription was active. The existing dumps were either extracted from Sega's B-Club! game rental service or ripped from the Game no Kanzume compilations released on the Mega-CD.
```
Comment (cuavas) [inline hash/megadriv.xml:63]: This line is definitely far too long – please wrap it at at a reasonable width (if in doubt, 80 columns should be OK).

File: hash/megadriv.xml [hunk 2]
```diff
@@ -25774,10 +25819,11 @@ Red screen
 		</part>
 	</software>
 
-	<software name="riddle">
-		<description>Riddle Wired (Japan, Sega Game Toshokan)</description>
+	<software name="riddle81">
+		<description>Riddle Wired - Hachigatsu Vol. 1 (Japan, Sega Game Toshokan)</description>
 		<year>1991</year>
 		<publisher>Sega</publisher>
+		<info name="alt_title" value="クイズ リドル･ワイアード ８月 VOL.1"/>
```
Comment (cuavas) [inline hash/megadriv.xml:25826]: Where does the Japanese title from?  I’ve seen it in some video titles as “クイズ リドル・ワイアード” (note the fulllwidth interpunct), but was it actually used officially?

The title screen says “RIDDLE WIRED クイズ独占企業の崩壊 8月 VOL. 1 (Riddle Wired Quiz Dokusen Kigyō no Hōkai 8-gatsu Vol. 1).

**Review by cuavas — COMMENTED** [review]
Where are the years from?  Are the years for the Game no Kanzume rips just the years that the Game no Kanzume collections themselves were released?

File: hash/megadriv.xml (hunk 1 above)
Comment (ICEknigh7) [reply hash/megadriv.xml:63]: Ok, done.

File: hash/megadriv.xml (hunk 2 above)
Comment (ICEknigh7) [reply hash/megadriv.xml]: They were given that full title in the menus of the Sega Game Toshokan service: https://www.youtube.com/watch?v=g0DHI_m9xIg
image
image

I've just added the full romanized and Japanese titles.

**Comment by ICEknigh7** [comment]
> Where are the years from?

Sega's website: https://www.sega.jp/history/hard/megadrive/software.html

> Are the years for the Game no Kanzume rips just the years that the Game no Kanzume collections themselves were released?

Yes, the years of those specific versions.

File: hash/megadriv.xml (hunk 2 above)
Comment (cuavas) [reply hash/megadriv.xml]: That’s far too open to interpretation though.  “クイズ リドル・ワイアード　メニュー” is the title of the submenu(s), and it also has the intermediate year/month groupings.  I think we can just use what’s on the title screen.

Comment (ICEknigh7) [reply hash/megadriv.xml]: Yeah, maybe they just wanted to give more prominence to the quiz category in the menu. I'll remove it as an alt.

Comment (cuavas) [reply hash/megadriv.xml]: Don’t worry, I’ll take care of it.

**Comment by cuavas** [comment]
BTW “Kandume” isn’t wrong per se, it’s just using Nihon-shiki romaji (intended to be relatively unambiguous) rather than the modified Hepburn romaji we tend to prefer in MAME.

**Pull request merged**